src/test/suite/ggshield-webview/gitguardian-quota-webview.test.ts

```diff
@@ -0,0 +1,101 @@
+import * as assert from "assert";
+import { GitGuardianQuotaWebviewProvider } from "../../../ggshield-webview/gitguardian-quota-webview";
+import { GGShieldConfiguration } from "../../../lib/ggshield-configuration";
+import {
+  ExtensionContext,
+  Memento,
+  Uri,
+  WebviewOptions,
+  WebviewView,
+} from "vscode";
+
+suite("GitGuardianQuotaWebviewProvider", () => {
+  let provider: GitGuardianQuotaWebviewProvider;
+  let mockGGShieldConfig: Partial<GGShieldConfiguration>;
+  let mockWebviewView: Partial<WebviewView>;
+  let mockWorkspaceState: Memento & {
+    setKeysForSync(keys: readonly string[]): void;
+  };
+  let mockContext: Partial<ExtensionContext>;
+
+  setup(() => {
+    mockWorkspaceState = {
+      get: (key: string) => undefined,
+      update: (key: string, value: any) => {
+        key = value;
+        return Promise.resolve();
+      },
+      keys: () => [],
+      setKeysForSync: (keys: readonly string[]) => {},
+    };
+
+    mockContext = {
+      workspaceState: mockWorkspaceState,
+    };
+    mockGGShieldConfig = {} as GGShieldConfiguration;
+
+    provider = new GitGuardianQuotaWebviewProvider(
+      mockGGShieldConfig as GGShieldConfiguration,
+      Uri.parse("file:///mock"),
+      mockContext as ExtensionContext
+    );
+
+    mockWebviewView = {
+      webview: {
+        html: "",
+        onDidReceiveMessage: () => ({ dispose: () => {} }),
+        cspSource: "",
+        options: {} as WebviewOptions,
+        postMessage: (message: any) => Promise.resolve(true),
+        asWebviewUri: (uri: Uri) => uri,
+      },
+      onDidChangeVisibility: () => ({ dispose: () => {} }),
+      visible: false,
+    };
+  });
+
+  test("should update the webview content when loading", () => {
+    provider["_view"] = mockWebviewView as WebviewView;
+    provider["isLoading"] = true;
+
+    provider["updateWebViewContent"]();
+
+    assert.ok(provider["_view"]?.webview.html.includes("<p>Loading...</p>"));
+  });
+
+  test("should display the quota when authenticated", () => {
+    provider["_view"] = mockWebviewView as WebviewView;
+    provider["isLoading"] = false;
+    provider["isAuthenticated"] = true;
+    provider["quota"] = 100;
+
+    provider["updateWebViewContent"]();
+
+    assert.ok(
+      provider["_view"]?.webview.html.includes("<p>Your current quota: 100</p>")
+    );
+  });
+
+  test("should display authentication prompt when unauthenticated", () => {
+    provider["_view"] = mockWebviewView as WebviewView;
+    provider["isLoading"] = false;
+    provider["isAuthenticated"] = false;
+
+    provider["updateWebViewContent"]();
+
+    assert.ok(
+      provider["_view"]?.webview.html.includes(
+        "<p>Please authenticate to see your quota.</p>"
+      )
+    );
+  });
+
+  test("should dispose the webview properly", () => {
```

Comment:
Is dispose ever called ? It's not on the WebviewViewProvider interface, I'm not really sure what it should do.

Reply:
I thought it was called in the main function activate() but I was wrong. Removed the test.